### PR TITLE
feat(SellerProducts): Seller can view their items in their collection #187364811

### PR DESCRIPTION
**Pull request by Sagastein**
## feat(SellerProducts): Seller can view their items in their collection #187364811

- Seller can view items or products in their collection
-  Sellers should only be able to see their products
- Only Authenticated sellers can view the items in their collection
-  Should be able to view all items in my collection
   such that I can inspect for deficiencies and other stock operations
  
 ## list of all products
 
image
 
   
 ## View product in  detail
image

## Loading products
image

## Product not found
image

### CODE REVIEW

**Review by shebz2023 — COMMENTED** [review]
Great work , just use some **components**

File: src/pages/vendorpages/ProductItem.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+              {new Date(product.createdAt).toLocaleDateString()}
+            </p>
+          </div>
+          <div className="flex gap-x-4">
+            <button className="bg-blue-600 hover:bg-blue-700 text-white px-6 py-2 rounded flex items-center transition duration-150 ease-in-out">
+              <FaEdit className="mr-2" /> Update
+            </button>
+            <button className="bg-red-600 hover:bg-red-700 text-white px-6 py-2 rounded flex items-center transition duration-150 ease-in-out">
+              <FaTrashAlt className="mr-2" /> Delete
+            </button>
```
Comment (shebz2023) [inline src/pages/vendorpages/ProductItem.tsx]: make a component for this

**Review by shebz2023 — DISMISSED** [review]
LGTM

**Review by tuyishimehope — DISMISSED** [review]
LGTM

File: src/components/vendorcomponents/Sidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  },
+  {
+    id: 4,
+    name: 'Categories',
+    icon: (
+      <MdOutlineCategory className="text-2xl" data-testid="categories-icon" />
+    ),
+    link: '/categories',
+  },
+];
```
Comment (SergeInGithub) [inline src/components/vendorcomponents/Sidebar.tsx]: These should be put in the data directory

File: src/pages/vendorpages/Product.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+              <th className="py-1 px-4 text-left">Name</th>
+
+              <th className="py-1 px-4 text-left">Price(Rwf)</th>
+              <th className="py-1 px-4 text-left">discount(%)</th>
+              <th className="py-1 px-4 text-left">Stock</th>
+              <th className="py-1 px-4 text-left">Categories</th>
+              <th className="py-1 px-4 text-left">Expire Date</th>
+              <th className="py-1 px-4 text-left">Action</th>
+            </tr>
+          </thead>
```
Comment (SergeInGithub) [inline src/pages/vendorpages/Product.tsx]: Same applies here

File: src/pages/vendorpages/Product.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+                      to={''}
+                      className="text-red-500 hover:text-red-700 focus:outline-none"
+                    >
+                      <FaTrashAlt className="text-xl" />
+                    </Link>
+                  </div>
+                </td>
+              </tr>
+            ))}
+          </tbody>
```
Comment (SergeInGithub) [inline src/pages/vendorpages/Product.tsx]: Same applies here

File: src/pages/vendorpages/Product.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+          </div>
+        </form>
+        <ActionButton
+          text="Add product"
+          colorClasses="bg-slate-800 text-white"
+          IconComponent={FaPlus}
+        />
+      </div>
+      <main>
+        <table className="min-w-full border bg-white">
```
Comment (SergeInGithub) [inline src/pages/vendorpages/Product.tsx:59]: Create a folder called tables under components and from there add Table component which will call TableHead, TableBody, etc...

File: src/types/Product.types.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
+export interface IProduct {
```
Comment (SergeInGithub) [inline src/types/Product.types.ts:1]: This is good. As usual, now let's create an index file that will handle exports from this directory

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Good work so far @Sagastein. Some reviews to be worked on

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Pull request merged**